### DISCUSSION TITLE
Close window on click-away

### DISCUSSION THREAD

**Discussion by LamprosPitsillos** [elkowar/eww#872] [Q&A]
I am looking for a way to close the calendar window when i click away from the window instead of having to click again to close it.
Is there such a way ?

**Comment by mantacid** [comment]
The absolute easiest (and consequentially idiotic) way I can think of would be to use a big window as your desktop background, and use an event box to close the calendar window when the desktop is clicked. But that has several problems, not the least of which is that this wouldn’t work if you clicked ANY OTHER WINDOW.

What I’d _actually_ suggest you do is make a script that uses something like `xdotool` (or a Wayland equivalent) to get the ID of the window under the mouse. Get that script to output the window ID when the mouse is clicked, and call that in a **listener** variable. Whenever the window ID changes, that variable will get updated. Now, you can use that variable in a conditional to set window visibility.

**Comment by w-lfchen** [comment]
I use something similar to what has already been suggested.
Instead of setting the window as the background, just put it on the top layer.
```
; generic closer widget
(defwidget closer [window]
  (button :onclick "eww close ${window} && eww close ${window}-closer"))

; closer window for the window "menu", works on wayland
(defwindow menu-closer
  :monitor 0
  :geometry (geometry :width "100%" :height "100%")
  :stacking "fg"
  :focusable false
  (closer :window "menu"))
```
If I now want to open the menu window, I just use `eww open-many menu-closer menu`.
If the closer window is not transparent, just add a class/`:style` with `all: unset;` and it should be transparent.
